x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -292,6 +291,8 @@ export class MapsAppView extends React.Component {
         globalState,
       }),
     };
+
+    if (newState.query) queryString.setQuery(newState.query);
```

Comment:
@nreese I had to add this line, setting the query to the manager in order to stabilize functional tests.
Is this addition correct and in the proper location in your opinion?

Reply:
This component needs some refactoring. This is a good place to put this until that refactoring occurs.